lib/beetle/base.rb

```diff
@@ -66,12 +66,14 @@ def queue(name)
           raise UnknownQueue.new("You are trying to bind a queue #{name} which is not configured!") unless opts
           logger.debug("Beetle: binding queue #{name} with internal name #{opts[:amqp_name]} on server #{@server}")
           queue_name = opts[:amqp_name]
-          creation_options = opts.slice(*QUEUE_CREATION_KEYS)
-          the_queue = nil
+          creation_keys = opts.slice(*QUEUE_CREATION_KEYS)
+
+          the_queue = declare_queue!(queue_name, creation_keys)
           @client.bindings[name].each do |binding_options|
             exchange_name = binding_options[:exchange]
             binding_options = binding_options.slice(*QUEUE_BINDING_KEYS)
-            the_queue = bind_queue!(queue_name, creation_options, exchange_name, binding_options)
+            logger.debug("Beetle: binding queue #{queue_name} to #{exchange_name} with opts: #{binding_options.inspect}")
+            the_queue.bind(exchange(exchange_name), binding_options)
```

Comment:
I think it would be better to introduce a method abstraction here. The code as is only works because luckily the method call for declaring a queue coincidentally looks identical between bunny and the amqp gem. I'm not sure that this will stay the same in the future, for example when we upgrade bunny.